src/RFC-0313_VNRegistration.md

```diff
@@ -0,0 +1,138 @@
+remove VNs whose registration UTXOs are older than the `VN re-registration` period, we can limit this. 
+
+> NOTE: An aggressive approach may be to force an anyone can spend script on the UTXO if the UTXO goes stale.
+
+In order to shuffle VNs inside of their respective VNCs, VNs MUST get a new VN-key each time they register. This is done to prevent some
+operators to do shard targeting attacks. 
+
+Each registration UTXO MUST contain the VN public key with a signature proving it knows the private key. 
+The signature MUST as a schnorr type signature as follows:
+$$
+\begin{aligned}
+sig = r + e*k \\\\
+e = hash(R || K || Commitment) \\\\
+\end{aligned}
+$$
+As the challenge, we hash the nonce, VN public key, and the commitment. We include the commitment to ensure each signature must be unique. This is important
+as we don't want users to reuse all signatures. 
+
+
+## VN-Key
+
+The VN-key is the unique key used to identify where in the [shard space] this key is generated unique each time the VN is registered to assign the
+VN to a different area of the [shard space] each time.
+
+The key MUST be calculated as follows:
+
+$$
+\begin{aligned}
+\text{VN-key} = hash(\text{VN-pubkey}||\text{blockhash})
+\end{aligned}
+$$
+Where the `blockhash` is the blockhash of the block, the registration transaction was mined in.
+It is essential we ensure that the `VN-Key` has some random entropy inside of it, as the `VN-Key` needs to be pseudorandom.
+
+## VN-Key option two
+
+We only need to ensure that the `VN-key` has some random entropy inside of it. At first glance, the `blockhash` is an excellent easy source of random entropy that's different
+for each block. But this causes some issues with the proof in the block. If the `blockhash` is dependent on all the information in the block, we cannot include the current VN Merkle proof in the current block. 
+
+An easy fix for this would be to use another source of random entropy that does not depend on the entire current block info. We can replace the `blockhash` with the
+`utxo Merkle root`. This is also pseudorandom but does not rely on the `VN key` calculation. 
+$$
+\begin{aligned}
+\text{VN-key} = hash(\text{VN-pubkey}||\text{utxo merkle root})
+\end{aligned}
+$$
+
+
+## UTXO de-registration
+An option exists to lock the registration UTXO for a certain amount of blocks before allowing spending to ensure registration stays open. But it can be argued that this
+serves no real purpose as we don't care if the UTXO exists or not. We care that we have an accurate list of active VNs. Therefore we can say that we allow VNs to 
+de0register themselves by spending their registration UTXOs to a normal UTXO. We will still have the [delayed view] of the main chain, so this de-registration action will
+only take effect after some time. 
+
+## Epochs
+Radix's Cerberus uses epochs to manage VN stability and shuffling of VNs inside the [shard space]. It can be argued that we don't need to group nodes inside of 
```

Comment:
```suggestion
Hotstuff and other BFT protocols use epochs to manage VN stability and shuffling of VNs inside the [shard space]. It can be argued that we don't need to group nodes inside of 
```